sqlx-core/src/net/tls.rs

```diff
@@ -75,12 +58,71 @@ where
             }
         };
 
+        #[cfg(feature = "_tls-rustls")]
+        let host = webpki::DNSNameRef::try_from_ascii_str(host)?;
+
         *self = MaybeTlsStream::Tls(connector.connect(host, stream).await?);
 
         Ok(())
     }
 }
 
+#[cfg(feature = "_tls-native-tls")]
+async fn configure_tls_connector(
+    accept_invalid_certs: bool,
+    accept_invalid_hostnames: bool,
+    root_cert_path: Option<&Path>,
+) -> Result<sqlx_rt::TlsConnector, Error> {
+    use sqlx_rt::native_tls::{Certificate, TlsConnector};
+
+    let mut builder = TlsConnector::builder();
+    builder
+        .danger_accept_invalid_certs(accept_invalid_certs)
+        .danger_accept_invalid_hostnames(accept_invalid_hostnames);
+
+    if !accept_invalid_certs {
+        if let Some(ca) = root_cert_path {
+            let data = fs::read(ca).await?;
+            let cert = Certificate::from_pem(&data)?;
+
+            builder.add_root_certificate(cert);
+        }
+    }
+
+    #[cfg(not(feature = "_rt-async-std"))]
+    let connector = builder.build()?.into();
+
+    #[cfg(feature = "_rt-async-std")]
+    let connector = builder.into();
+
+    Ok(connector)
+}
+
+#[cfg(feature = "_tls-rustls")]
+async fn configure_tls_connector(
+    _accept_invalid_certs: bool,
+    _accept_invalid_hostnames: bool,
+    root_cert_path: Option<&Path>,
+) -> Result<sqlx_rt::TlsConnector, Error> {
+    // FIXME: Support accept_invalid_certs / accept_invalid_hostnames
```

Comment:
I think it wouldn't be too hard to implement accepting *any* certs, but turning off only host validation seems potentially harder. Also all of this is not my area of expertise so I would prefer if this could be merged with the rustls TLS backend just not handling these options.

Reply:
We do need to be able to turn off validation because for testing we need to be able to accept the self-signed certificates that Postgres generates on startup. That would be both accepting invalid certs and invalid hostnames together though.

Reply:
I'll try integrating @BlackHoleFox'es changes.